sched/sched/sched_lock.c

```diff
@@ -171,18 +131,17 @@ int sched_lock(void)
            * locked (or we would not be executing!).
            */
 
-          spin_setbit(&g_cpu_lockset, this_cpu(), &g_cpu_locksetlock,
-                      &g_cpu_schedlock);
+          DEBUGASSERT((g_cpu_lockset & (1 << cpu)) == 0);
+          g_cpu_lockset |= (1 << cpu);
         }
       else
         {
           /* If this thread already has the scheduler locked, then
-           * g_cpu_schedlock() should indicate that the scheduler is locked
+           * g_cpu_lockset should indicate that the scheduler is locked
            * and g_cpu_lockset should include the bit setting for this CPU.
            */
 
-          DEBUGASSERT(spin_is_locked(&g_cpu_schedlock) &&
-                      (g_cpu_lockset & (1 << this_cpu())) != 0);
+          DEBUGASSERT((g_cpu_lockset & (1 << cpu)) != 0);
         }
```

Comment:
Look like scheduler lock can be token by two cpus?

Reply:
> Look like scheduler lock can be token by two cpus?

yes, The current implementation of sched_lock is that whenever any CPU acquires the sched_lock, it causes the sched_lock to be in effect for all CPUs. There will be subsequent patchs to address this issue.

Reply:
Not related performance issue.
If there is any resource protected by `g_cpu_schedlock` but we can get scheduler lock with different cpus simultaneously we will face race issue.
If I am wrong, please correct me.

Reply:
This is a common misunderstanding about sched_lock. It cannot achieve mutual exclusion of resources, especially in SMP scenarios.
In most cases, if we want protect resources, enter_critical_section should be used.